### PR TITLE
Add golangci lint check on pull requests

### LINKED ISSUE #620: Add golangci-lint
* Cover all non test Go files
* Run in CI

### PR DESCRIPTION
**Pull request by fkneeland-figure**
Resolves #620 
I added a github action to run the golangci lint test on all non-test golang files when a pr is created.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -60,6 +60,17 @@ jobs:
           paths:
             - ".git"
 
+  lint:
```
Comment (ethanfrey) [inline .circleci/config.yml:63]: Looks good.

I was surprised it didn't show up on the PR. But then realised it is missing a section like this: https://github.com/CosmWasm/wasmd/pull/645/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L194-L196

Can you add 3 lines to workflow (running after setup-dependencies makes sense to me), and I can see the results in the CI.

File: Makefile [hunk 2]
```diff
@@ -146,8 +146,8 @@ test-sim-multi-seed-short: runsim
 ###############################################################################
 
 lint:
-	golangci-lint run
-	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" | xargs gofmt -d -s
+	golangci-lint run --tests=false
```
Comment (ethanfrey) [inline Makefile:149]: Good point. We can lint tests later with less urgency.

File: x/wasm/ibc.go [hunk 3]
```diff
@@ -34,6 +35,7 @@ func (i IBCHandler) OnChanOpenInit(
 	counterParty channeltypes.Counterparty,
 	version string,
 ) error {
+	  fmt.Println("hi")
```
Comment (ethanfrey) [inline x/wasm/ibc.go:38]: Please remove before merging. (But fine for testing CI)

**Comment by fkneeland-figure** [comment]
I added all the changes from running `make format` into this pr, I am cool with those changes being merged in this pr or separately in the other pr whatever is prefered.

**Comment by ethanfrey** [comment]
Personally, I would revert the last 2 commits (or move them to another PR on top of this).

I could merge the CI changes alone (that is a clear unit) and then there can be multiple "cleanup" PRs.

**Comment by ethanfrey** [comment]
Can you ping me when you revert those 2 and I can merge

**Comment by fkneeland-figure** [comment]
This is now passing ci and is ready for review once https://github.com/CosmWasm/wasmd/pull/648 has been merged into master and then master is merged into this pr

File: x/wasm/types/genesis.go [hunk 4]
```diff
@@ -1,6 +1,5 @@
 package types
 
-import "C"
```
Comment (alpe) [inline x/wasm/types/genesis.go:3]: 👍

File: x/wasm/types/types.pb.go [hunk 5]
```diff
@@ -140,35 +141,35 @@ type AccessConfig struct {
 	Address    string     `protobuf:"bytes,2,opt,name=address,proto3" json:"address,omitempty" yaml:"address"`
 }
 
-func (m *AccessConfig) Reset()         { *m = AccessConfig{} }
-func (m *AccessConfig) String() string { return proto.CompactTextString(m) }
+func (a *AccessConfig) Reset()         { *a = AccessConfig{} }
```
Comment (alpe) [inline x/wasm/types/types.pb.go:144]: Please do not modify generated `*pb.go` files beside automated formatting or goimports. This will be overwritten by the next generator run and causes a lot confusion in code reviews.
Please revert the changes

File: .golangci.yml [hunk 6]
```diff
@@ -1,25 +1,26 @@
...
+  tests: false
+
 linters:
+  disable-all: true
   enable:
     - bodyclose
     - deadcode
     - depguard
     - dogsled
-    - errcheck
```
Comment (alpe) [inline .golangci.yml:7]: can you elaborate on the checks that you added/ removed?
